boards/beagle/pocketbeagle_2/pocketbeagle_2_am6232_m4.dts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2025 Ayush Singh, BeagleBoard.org Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <ti/am62x_m4.dtsi>
+#include "pocketbeagle_2_am6232_m4-pinctrl.dtsi"
+
+/ {
+	model = "BeagleBoard.org PocketBeagle 2";
+	compatible = "beagle,pocketbeagle_2_m4";
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,ipc_shm = &ddr0;
```

Comment:
We have mailbox support now, you should be able to add the following here: `zephyr,ipc = &ipc0;`

Then since all the OpenAMP/RPMSG stuff still uses the old IPM, add an adapter node like this below,

```
	ipc0: ipc {
		compatible = "zephyr,mbox-ipm";
		mboxes = <&mbox0 0>, <&mbox0 1>;
		mbox-names = "tx", "rx";
	};
```

Should allow for out-of-box OpenAMP demos to work. See [SK-AM64](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/ti/sk_am64/sk_am64_am6442_m4.dts#L62) for an example.

Reply:
Done